code/parachain/frame/composable-maths/src/dex/tests.rs

```diff
@@ -0,0 +1,155 @@
+use crate::dex::constant_product::compute_in_given_out_new;
+use proptest::prelude::*;
+use sp_runtime::{ArithmeticError, DispatchError, Permill};
+
+/// Tests related to constant product math functions
+mod constant_product {
+	use super::*;
+	/// Tests related to the function `compute_in_given_out_new`
+	mod compute_in_given_out_new {
+		use super::*;
+
+		/// should always be equal to the number of elements in the `list` array found in
+		/// `checked_inputs_and_outputs` function
+		const CHECKED_I_AND_O_SIZE: u32 = 5;
```

Comment:
If the list defined in the following `prop_compose!` were defined as a `const` outside of the fn, then you could just use `LIST.len()` here